src/controllers/update/update.manage.trusts.tell.us.about.the.former.bo.controller.ts

```diff
@@ -0,0 +1,127 @@
+import { saveAndContinue } from '../../utils/save.and.continue';
+import { mapBeneficialOwnerToSession } from '../../utils/trust/historical.beneficial.owner.mapper';
+import { getTrustInReview, getTrustee, getTrusteeIndex } from '../../utils/update/review_trusts';
+import { yesNoResponse } from '../../model/data.types.model';
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const appData = getApplicationData(req.session);
+    const trust = getTrustInReview(appData) as Trust;
+
+    // for editing existing trustee
+    const trusteeId = req.params[ROUTE_PARAM_TRUSTEE_ID];
+    const formData = trusteeId ? mapFormerBOFromModelToForm(trust, trusteeId) : {} as TrustHistoricalBeneficialOwnerForm;
+
+    const pageProperties = getPageProperties(trust, formData);
+
+    return res.render(UPDATE_MANAGE_TRUSTS_TELL_US_ABOUT_THE_FORMER_BO_PAGE, pageProperties);
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const post = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const session = req.session as Session;
+    const appData = getApplicationData(session);
+    const trust = getTrustInReview(appData) as Trust;
+    const trusteeId = req.params[ROUTE_PARAM_TRUSTEE_ID];
+    const formData: TrustHistoricalBeneficialOwnerForm = req.body as TrustHistoricalBeneficialOwnerForm;
+
+    // check for form validation errors
+    const errorList = validationResult(req);
+    if (!errorList.isEmpty()) {
+      const pageProperties = getPageProperties(trust, formData, formatValidationError(errorList.array()));
+      return res.render(UPDATE_MANAGE_TRUSTS_TELL_US_ABOUT_THE_FORMER_BO_PAGE, pageProperties);
+    }
+
+    const trustee = mapBeneficialOwnerToSession(req.body);
+    const trusteeIndex = getTrusteeIndex(trust, trusteeId, TrusteeType.HISTORICAL);
+
+    if (trust.HISTORICAL_BO && trusteeIndex >= 0) {
+      // update existing trustee
+      trust.HISTORICAL_BO[trusteeIndex] = trustee;
+    } else {
+      // add new trustee
+      trust.HISTORICAL_BO?.push(trustee);
+    }
+
+    setExtraData(session, appData);
+    await saveAndContinue(req, session, true);
+
+    return res.redirect(UPDATE_MANAGE_TRUSTS_REVIEW_FORMER_BO_URL);
+  } catch (error) {
+    next(error);
+  }
+};
+
+const getPageProperties = (trust: Trust, formData?: TrustHistoricalBeneficialOwnerForm, errors?: FormattedValidationErrors) => ({
+  backLinkUrl: UPDATE_MANAGE_TRUSTS_REVIEW_FORMER_BO_URL,
+  templateName: UPDATE_MANAGE_TRUSTS_TELL_US_ABOUT_THE_FORMER_BO_PAGE,
+  pageParams: {
+    title: "Tell us about the former beneficial owner",
+  },
+  pageData: {
+    trustData: {
+      trustId: trust.trust_id,
+      trustName: trust.trust_name
+    },
+    trusteeType: TrusteeType,
+  },
+  formData,
+  errors
+});
+
+const mapFormerBOFromModelToForm = (
```

Comment:
We can save on this duplication actually, check Conuir's PR

Reply:
done, thanks!